test/junit/scala/tools/nsc/reporters/ConsoleReporterTest.scala

```diff
@@ -0,0 +1,188 @@
+package scala
+package tools.nsc
+package reporters
+
+import java.io.{ByteArrayOutputStream, StringReader, BufferedReader, PrintStream, PrintWriter}
+import org.junit.Assert._
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+import scala.tools.nsc.Settings
+import scala.tools.nsc.reporters.ConsoleReporter
+import scala.reflect.internal.util._
+import scala.reflect.io.AbstractFile
+
+
+@RunWith(classOf[JUnit4])
+class ConsoleReporterTest {
+    val noPos = NoPosition
+    val source = "Test_ConsoleReporter"
+    val batch_file = new BatchSourceFile(source, "For testing".toList)
+    val posWithSource = new OffsetPosition(batch_file, 4)
+    val content = posWithSource.lineContent
+
+    val err_out = new ByteArrayOutputStream  //for storing the output printed in the ConsoleReporter
+    val output_writer = new PrintWriter(err_out)
+    val input = new StringReader("s\n")
+    val input_reader = new BufferedReader(input)
+    val setting = new Settings()
+    setting.maxerrs.tryToSet(List("50"))  //changing maxerrs
+    setting.maxwarns.tryToSet(List("25")) //chaging maxwarns
+    val reporter = new ConsoleReporter(setting, input_reader, output_writer)
+
+    @Test
+    def printMessageTest(): Unit = {
+      reporter.printMessage("Hello World!")
+      assertEquals("Hello World!\n", err_out.toString)
+      err_out.reset
```

Comment:
You could capture this repeated pattern in a routine that performs the assert and the reset. I often see simple helpers like that in unit tests.